src/ui/components/carousel/basic_carousel_component.py

```diff
@@ -7,54 +7,39 @@
 
 from src.ui.components.base_component import BaseComponent
 
-LEFT_ARROW_BUTTON = (By.XPATH, ".//span[contains(@aria-label, 'arrow-left')]")
-RIGHT_ARROW_BUTTON = (By.XPATH, ".//span[contains(@aria-label, 'arrow-right')]")
-SLICK_DOTS = (By.XPATH, ".//ul[contains(@class,\"slick-dots\")]/li")
-SLICK_DOTS_CONTAINER = (By.XPATH, ".//ul[contains(@class,\"slick-dots\")]")
-SLIDER_CONTAINER = (By.XPATH, ".//div[contains(@class,\"slick-slider\")]")
-
-
 class BasicCarouselComponent(BaseComponent):
     def __init__(self, driver: webdriver, node: WebElement) -> None:
         super().__init__(driver)
         self._driver = driver
         self._node = node
-        self._left_arrow_button = None
-        self._right_arrow_button = None
-        self._slick_dots = None
-        self._slick_dots_container = None
-        self._slider_container = None
+        self.locators = {
+            "left_arrow_button": ("xpath",".//span[contains(@aria-label, 'arrow-left')]"),
+            "right_arrow_button": ("xpath", ".//span[contains(@aria-label, 'arrow-right')]"),
+            "slick_dots": ("xpath", ".//ul[contains(@class,\"slick-dots\")]/li"),
+            "slick_dots_container": ("xpath", ".//ul[contains(@class,\"slick-dots\")]"),
+            "slider_container": ("xpath", ".//div[contains(@class,\"slick-slider\")]"),
+        }
         self._wait = WebDriverWait(self._driver, 30)
 
     @property
     def left_arrow_button(self) -> WebElement:
-        if not self._left_arrow_button:
-            self._left_arrow_button = self._node.find_element(*LEFT_ARROW_BUTTON)
-        return self._left_arrow_button
+        return self._node.find_element(*self.locators["left_arrow_button"])
```

Comment:
We don't need to use find_element for web elements and write getters for them. PageFActory will create getter using key from locators dict. We only need getters for list of web elements)